### PR TITLE
Add Snyk checks to CI

### PR DESCRIPTION
**Pull request by isavov**
**Description**:
This PR integrates and extends the Snyk gradle plugin to add Snyk scanning task to the gradle setup and integrate it in CI.
Snyk Open Source and Snyk Code scanning are implemented for pull requests and the monitored snapshot is updated on merge to main.

* Add gradle plugin
* Add jobs to the Security action

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/workflows/security.yml [hunk 1]
```diff
@@ -63,3 +63,94 @@ jobs:
...
+      - name: Execute Gradle Snyk Code Test
+        uses: gradle/gradle-build-action@v2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        with:
+          arguments: snyk-code
+
+      - name: Publish Snyk Results
+        if: ${{ !cancelled() }}
```
Comment (nathanklick) [inline .github/workflows/security.yml:126]:
```suggestion
        if: ${{ !cancelled() && always() }}
```

File: .github/workflows/security.yml [hunk 2]
```diff
@@ -63,3 +63,94 @@ jobs:
...
+      - name: Execute Gradle Snyk Test
+        uses: gradle/gradle-build-action@v2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        with:
+          arguments: snyk-test
+
+      - name: Publish Snyk Results
+        if: ${{ !cancelled() }}
```
Comment (nathanklick) [inline .github/workflows/security.yml:93]:
```suggestion
        if: ${{ !cancelled() && always() }}
```

**Review by steven-sheehy — CHANGES_REQUESTED** [review]
Verified the workflow ran: https://github.com/hashgraph/hedera-mirror-node/actions/runs/4419597443

File: .github/workflows/security.yml [hunk 3]
```diff
@@ -63,3 +63,94 @@ jobs:
...
+        uses: actions/checkout@v3
+
+      - name: Install JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: 17
+
+      - name: Setup Snyk
+        run: npm install -g snyk snyk-to-html @wcj/html-to-markdown-cli
```
[the remark this answers is not in the document]
Comment (isavov) [reply .github/workflows/security.yml]: I think it shouldn't be too different, as the gradle plugin checks for the presence of a system wide install of snyk and if there isn't one it downloads it in the project's dir. Anyways this is a legacy step from a previous version of the implementation where I tried to run a separate snyk scan and generate a separate report for each project within mirror-node (where a separate snyk binary would be downloaded for each) and is not needed anymore, so I'm removing it.

**Comment by steven-sheehy** [comment]
DCO check failing

**Comment by isavov** [comment]
> DCO check failing

Fixed and reviews addressed

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**